### ISSUE TITLE
Inability to subset old .grid.nc mesh files

### ISSUE THREAD

**Issue by MiCurry** [MiCurry/MPAS-Limited-Area#28]
The following meshes (which are available to download as of 6/27/2019) are not able to run with MPAS-Limited-Area:

- `x1.4002.grid.nc`
- `x1.1024002.grid.nc`
- `x1.5898242.grid.nc`
- `x1.535554.gric.nc`
- `x4.655362.grid.nc`
- `x4.999426.grid.nc`

If you are getting either a `MemoryError` or an `IndexError` with these meshes, then you most likely have an older mesh that is not compatible with MPAS-Limited-Area. These older meshes contain empty initialization variables and they also mark their garbage cells/edges/vertices with an undefined index, hence the garbage cell.

We are currently in the process of updating these meshes without these extra varariables; however, if you wish to use these method please see the post below for two workarounds.

---

Here is an example of the errors from these old meshes:

``` Python
Copying variable u... Traceback (most recent call last):
  File "/glade/u/home/mcurry/mpas/MPAS-Limited-Area/tests/test_meshes.py", line 67, in test_grids
    region, regionGraph = regional_area.gen_region()
  File "/glade/u/home/mcurry/mpas/MPAS-Limited-Area/limited_area/limited_area.py", line 153, in gen_region
    **kwargs)
  File "/glade/u/home/mcurry/mpas/MPAS-Limited-Area/limited_area/mesh.py", line 317, in subset_fields
    region.mesh.variables[var][:] = arrTemp[glbBdyEdgeIDs]
  File "/gpfs/u/apps/dav/opt/python/3.6.8/gnu/7.3.0/pkg-library/20190326/lib/python3.6/site-packages/numpy/ma/core.py", line 3174, in __getitem__
    dout = self.data[indx]
IndexError: index 1439492 is out of bounds for axis 0 with size 1
```

``` Python
  File "/glade/u/home/mcurry/mpas/MPAS-Limited-Area/tests/test_meshes.py", line 67, in test_grids
    region, regionGraph = regional_area.gen_region()
  File "/glade/u/home/mcurry/mpas/MPAS-Limited-Area/limited_area/limited_area.py", line 153, in gen_region
    **kwargs)
  File "/glade/u/home/mcurry/mpas/MPAS-Limited-Area/limited_area/mesh.py", line 317, in subset_fields
    region.mesh.variables[var][:] = arrTemp[glbBdyEdgeIDs]
  File "/gpfs/u/apps/dav/opt/python/3.6.8/gnu/7.3.0/pkg-library/20190326/lib/python3.6/site-packages/numpy/ma/core.py", line 3174, in __getitem__
    dout = self.data[indx]
MemoryError
```

**Comment by MiCurry** [comment]
The problem with these meshes is that they contain a few extra variables which new meshes do not. We are currently in the processing of updating these old meshes to not contain these extra variables: 
```
fEdge
fvertex
h_s
u
v
h
vh
circulation
vorticity
ke
tracers
```

There are currently two work arounds: 

# Work-Around 1 - Remove extra variables from the grid.nc file

Remove the extra variables using NCO's kitchen sink command:

```
ncks -x -v fEdge,fvertex,h_s,u,v,h,vh,circulation,vorticity,ke,tracers x1.1024002.grid.nc x1.1024002.grid.fix.nc
```
and then run create_region.

# Work-Around 2 - Run Static Interp on entire grid domain

A probably easier and a method that will probably save you more time, is to run the static interpolation upon the global grid.nc file, and then running create_region on the static.nc file. Doing so will remove the extra variables and will eliminate the need to run the static interp upon regions (in case you need a few tries to get it right).

**Comment by MiCurry** [comment]
Closed by #33
